Library/Homebrew/brew.sh

```diff
@@ -3,15 +3,27 @@
 ##### to be able to `source` in shell configurations run quickly.
 #####
 
-# Doesn't need a default case because we don't support other OSs
+# Doesn't need a default case because we don't support other OSs/architectures
 # shellcheck disable=SC2249
-HOMEBREW_PROCESSOR="$(uname -m)"
-HOMEBREW_PHYSICAL_PROCESSOR="${HOMEBREW_PROCESSOR}"
-HOMEBREW_SYSTEM="$(uname -s)"
-case "${HOMEBREW_SYSTEM}" in
-  Darwin) HOMEBREW_MACOS="1" ;;
-  Linux) HOMEBREW_LINUX="1" ;;
+case "${MACHTYPE}" in
+  arm64-*)
+    HOMEBREW_PROCESSOR="arm64"
+    ;;
+  x86_64-*)
+    HOMEBREW_PROCESSOR="x86_64"
+    ;;
+esac
+case "${MACHTYPE}" in
+  *apple-darwin*)
+    HOMEBREW_SYSTEM="darwin"
+    HOMEBREW_MACOS="1"
+    ;;
+  *linux*)
+    HOMEBREW_SYSTEM="linux"
+    HOMEBREW_LINUX="1"
```

Comment:
`HOMEBREW_SYSTEM` used to have an uppercase first character. Probably worth keeping that way to avoid subtle changes to bottle tabs and analytics.

Reply:
Fixed.